### ISSUE TITLE
Possible axis rendering regression in 35.6.4 or 35.7.0

### ISSUE THREAD

**Issue by pushred** [FormidableLabs/victory#1861]
# Bugs and Questions

### Checklist

- [x] This is not a `victory-native` specific issue. (Issues that only appear in `victory-native` should be opened [here](https://github.com/FormidableLabs/victory-native/issues/new))

- [x] I have read through the [FAQ](https://formidable.com/open-source/victory/docs/faq) and [Guides](https://formidable.com/open-source/victory/guides/) before asking a question

- [ ] I am using the latest version of Victory

- [x] I've searched open issues to make sure I'm not opening a duplicate issue

### The Problem

We have been pinned to Victory 35.1.1 (but all sub-dependencies are installing the latest version) and a previously implemented chart is no longer rendering correctly after some recent change. I suspected #1835 was to blame but in my repro it's actually 35.6.4 where I first see the issue.

The chart in question is using some custom scales using d3-scale and custom ticks with d3-axis, so not sure if this is where things are going wrong or whether that usage is even supported. But since there appears to have been a breaking change somewhere I wanted to reach out and see if the issue is valid and whether we're doing something especially wrong!

To workaround I am currently using Yarn resolutions to pin everything to the exact earlier version. I believe I'll have to do this going forward regardless if we wish to control what version we're on. I'm assuming that the package versions for Victory and all of it's sub-packages must be fixed to the same version to work as intended.

### Reproduction

https://codesandbox.io/s/green-browser-p9h2i

**Comment by topched** [comment]
Not the exact same issue but I just upgraded from 35.5.1 to 35.8.2 and I am noticing issues with axis rendering -- specifically when it comes to yOffset. Not sure if I should create a separate issue or piggy back on this one.

Here is a reproduction
https://codesandbox.io/s/unruffled-sammet-y7s10?file=/index.js

Changed the version of victory to the latest version and notice that bottom axis is no longer aligned to the bottom its offset from 0. Changing back to 35.5.1 and the axis is correctly aligned at the bottom
